### ISSUE TITLE
Option to limit download/upload speed.

### ISSUE THREAD

**Issue by MoOx** [webtorrent/webtorrent-desktop#435]
That would be nice to have this kind of options (especially for slow connection).
Most torrent client have this kind of options.

**Comment by quicoto** [comment]
This feels like a must feature. Otherwise when starting a torrent it takes the whole network bandwidth killing the connection for other devices or computers connected to it.

Thank you very much

**Comment by mikeytusa** [comment]
Along the same lines, if it were possible to limit the amount of connections being made this would be awesome.  This thing crushes my router on my DSL internet.  Disconnects constantly due to too many connections being made.

**Comment by feross** [comment]
@mikeytusa There is already a way to limit the amount of connections. Use the `maxConns` property. :-)

**Comment by demoneaux** [comment]
@feross not sure if you realised, but this is the WebTorrent Desktop repo, and we don't have an option for that yet. I'll try to see if I can add one soon :)

**Comment by feross** [comment]
@demoneaux Lol, thanks for the catch :)

**Comment by BrandTurner** [comment]
I love the app, but it's been over a year and there's still no ability to limit the speeds. Running webtorrent maxes out my connection speed and simply makes any other application that makes network requests unusable on my entire network. 

Right now, I tried to download a MOOC that was offered via torrent while also working on a side project. It's really disappointing that this has been a known issue for a year and has not been resolved. Nothing complex, just a user controlled global limit to D/L and U/L so that I can use it as both a traditional torrent client and a streaming torrent player. Don't mind the ratios because this app is meant to be on a media machine, not a seedbox.

This simple change would make WebTorrent a functional torrent client, allowing me to only have torrent app installed. If help is needed, I'd be happy to volunteer my time on this feature (I need to start contributing to OSS instead of complaining :)

**Comment by MoOx** [comment]
> I need to start contributing to OSS instead of complaining

Exactly. A PR is always welcome in OSS :)

**Comment by Flet** [comment]
It looks like there is a dependency on `webtorrent` itself to enable upload/download throttling before we can implement it here.

Relevant `webtorrent` issue: webtorrent/webtorrent#163
And an outstanding PR that seems to implement it: webtorrent/webtorrent#1040

**Comment by coleplx** [comment]
I'm so waiting for this too...

**Comment by chuanxd** [comment]
https://github.com/webtorrent/webtorrent/pull/1040

**Comment by raxim1** [comment]
maybe b=AS:64 (64KB) to set in SDP?

**Comment by skzap** [comment]
Seems like its working in webtorrent now :)

**Comment by jimmywarting** [comment]
A very much needed feature. I can't do much else when webtorrent steals the hole network. Sometimes i need to reconnect to my wifi.

**Comment by anasshm** [comment]
2 Years later still nothing . the app is really unusable for me right now .

**Comment by anasshm** [comment]
Guys try to tweet at them on twitter @WebTorrentApp , i see that they are active there hopefully they can do something about it .

**Comment by feross** [comment]
👋 Hey, thanks for your passion for fixing this issue. This project is a volunteer-run effort. I'm slowly going through my backlog and trying to get a sense of what's important to the future of the WebTorrent project.

This issue will be high on my list. But realize that there are only a handful of us working on WebTorrent and this isn't our full-time jobs. If you have JavaScript skills, please consider helping out by fixing bugs, leaving comments to help users who are asking questions, etc. Anything that can help lighten the load of the maintainers is appreciated.

Thanks.

**Comment by markg85** [comment]
+1 to this. I'm also facing the annoyance of a torrent downloading at a few kb at most (if any at all) and sucking up my upload bandwidth completely.

And while this project looks very cool, contributing a change like this myself (i am a developer) has a bit of a steep learning curve as the code logic isn't quite clear to me. For instance, where is the part that is actually downloading a torrent? I'm still searching...

**Comment by sotiriskoletsios** [comment]
A must future...

**Comment by jeedee23** [comment]
NETLIMITER.COM

**Comment by EvertonQuadros** [comment]
> 
> 
> 2 Years later still nothing . the app is really unusable for me right now .

i agree with you, it seems to be a great potential but in current state is pretty unusable, unfortunately.

**Comment by 2020karaulov2020** [comment]
Please add fuсКing speed limit option)) I can't watch videos because upload speed bigger than download))

**Comment by 2020karaulov2020** [comment]
@alxhotel @bnjmnt4n @DiegoRBaquero @feross @Flet @imsnif @iovis @josephfrazier @mappum @mvayngrib @rom1504 @SilentBot1 @transitive-bullshit @ungoldman @watson 
![9FABhjW 1](https://user-images.githubusercontent.com/62130676/110248886-57c6f780-7f84-11eb-9284-cff1d9e001bf.png)

**Comment by 2020karaulov2020** [comment]
@alxhotel @bnjmnt4n @DiegoRBaquero @feross @Flet @imsnif @iovis @josephfrazier @mappum @mvayngrib @rom1504 @SilentBot1 @transitive-bullshit @ungoldman @watson
**Please add fuсКing speed limit option)) I can't watch videos because upload speed bigger than download))**
![110248886-57c6f780-7f84-11eb-9284-cff1d9e001bf](https://user-images.githubusercontent.com/62130676/110269870-acdf2980-7fd5-11eb-84bc-b2c76967df0d.png)

**Comment by Zibri** [comment]
any news on this?

**Comment by drakonkat** [comment]
I think they have added it time ago, you can check here ![image](https://user-images.githubusercontent.com/17176640/161543736-9b628888-b8a3-420c-86d5-5359c226ffd9.png)
and at this link of updated docs "https://github.com/webtorrent/webtorrent/blob/master/docs/api.md"

**Comment by Zibri** [comment]
Yep.. I meant in the GUI...

**Comment by dtantono** [comment]
I saw the screenshot. how to do that?

**Comment by jeedee23** [comment]
not not relevant anymore
best regards
Johan

Op vr 4 nov. 2022 om 13:07 schreef github-actions[bot] <
***@***.***>:

**Comment by jeedee23** [comment]
Yes, it is working.
Met vriendelijke groeten, meileures salutations,mit freundlichen Grüßen,Best
regards, ودي الخاص


*Johan Degraeve*

Admin data:

Company:-----------------------*Aanscharius Vacuum & Dust Solutions*
Address:-----------------------Veenstraat 28 | 8470 Gistel | Belgium
Phone (main+mob)---------------+3259250944 |+32489947935
BTW/TVA/VAT/MSWT: ------------ BE0841892791
RC:----------------------------2.205.300.245
IBAN/Swift---------------------BE41 7380 3449 2010 / KREDBEBB


Op za 7 jan. 2023 om 13:05 schreef github-actions[bot] <
***@***.***>:

**Comment by maddymodd** [comment]
I'm currently working on implementing this.

**Comment by maddymodd** [comment]
Completed. The PR is live: https://github.com/webtorrent/webtorrent-desktop/pull/2308

![](https://user-images.githubusercontent.com/121954045/211311718-dc95c716-4729-417e-bd59-7abc07ad1cd5.png)

**Comment by maddymodd** [comment]
Submitted a new PR that addresses this feature at https://github.com/webtorrent/webtorrent-desktop/pull/2310

**Comment by maddymodd** [comment]
> Is this still relevant? If so, what is blocking it? Is there anything you can do to help move it forward?

Yes, it is still relevant. What's blocking it is that the fix at https://github.com/webtorrent/webtorrent-desktop/pull/2310 needs at least 1 approving review.

**Comment by alxhotel** [comment]
Lets try to get https://github.com/webtorrent/webtorrent-desktop/pull/2310 merged

**Comment by jgabriel98** [comment]
> Is this still relevant? If so, what is blocking it? Is there anything you can do to help move it forward?

yes
